### PR TITLE
🎯 Math Magicians | Components 🎯

### PR DESCRIPTION
**Pull request by Rysth**
# 🏅 Features Implemented
_Project Requirements -> https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_components.md_

List of things that were implemented based on the requirements:

## 📝 General requirements

- ✅ No Linters errors.
- ✅ Used GitFlow correctly.
- ✅ Created a professional `README` file.

## 📝 JavaScript requirements

- ✅ Followed the best JavaScript practices.

## 📝 Project requirements

- ✅ Deleted all the boilerplate from Create React App (text, images, styles).
- ✅ Created a new directory called components.
- ✅ Created a new Calculator.js file within the new folder.
- ✅ Created a React component that matches this design.
- ✅ Implemented the design only (JSX and CSS).
- ✅ Used **functional** components.
- ✅ Created at least 1 child component.
- ✅ Imported and used within the top-level component called `App.js`.
- ✅ Rendered without problems by using `npm start` in the console.

## 📷 Screenshots

image

_John Palacios_
_Microverse Student_

### CODE REVIEW

File: src/reportWebVitals.js [hunk 1]
```diff
@@ -1,15 +0,0 @@
-const reportWebVitals = (onPerfEntry) => {
```
Comment (julie-ify) [inline src/reportWebVitals.js:1]: - [x] Nice work removing this file, but you still need to remove these other files in the public directory.

  image

File: src/components/Calculator.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+CalculatorButton.defaultProps = {
+  content: '',
+  classValue: '',
+};
+
+CalculatorButton.propTypes = {
+  content: PropTypes.string,
+  classValue: PropTypes.string,
+};
```
Comment (julie-ify) [inline src/components/Calculator.js]: - [x] **[OPTIONAL]** It is often the case that we have one file for one component. I suggest you move these three components into 3 different files and import them for use instead in this Calculator component.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Rysth,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/reportWebVitals.js (hunk 1 above)
Comment (Rysth) [reply src/reportWebVitals.js:1]: Done!

File: src/components/Calculator.js (hunk 2 above)
Comment (Rysth) [reply src/components/Calculator.js]: Done!

**Review by julie-ify — APPROVED** [review]
Hi @Rysth,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**